audittools/auditor.go

```diff
@@ -0,0 +1,255 @@
+/*******************************************************************************
+*
+* Copyright 2024 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+package audittools
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sapcc/go-api-declarations/cadf"
+	"github.com/sapcc/go-bits/assert"
+	"github.com/sapcc/go-bits/logg"
+	"github.com/sapcc/go-bits/osext"
+)
+
+// Auditor is a high-level interface for audit event acceptors.
+// In a real process, use NewAuditor() or NewNullAuditor() depending on whether you have RabbitMQ client credentials.
+// In a test scenario, use NewMockAuditor() to get an assertable mock implementation.
+type Auditor interface {
+	Record(EventParameters)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// type standardAuditor
+
+// AuditorOpts contains options for NewAuditor().
+type AuditorOpts struct {
+	// Required. Identifies the current process.
+	// The ID field should be set to a UUID, such as those generated by GenerateUUID().
```

Comment:
Which ID field? The one inside of observer?

Reply:
Yes. I clarified the comment.